api/python/tests/integration/test_packages.py

```diff
@@ -666,6 +672,70 @@ def test_s3_set_dir(self):
 
             list_object_versions_mock.assert_called_with('bucket', 'foo/')
 
+    def test_set_dir_update_policy(self):
+        """Verify building a package with update policy. """
+
+        # Test for local src dir
+        path = pathlib.Path(__file__).parent
+        data_dir = pathlib.Path(path, "data")
+        pkg = Package()
+        pkg = pkg.set_dir("/", data_dir, meta={'name': 'test_meta'})
+        assert 'nested' in pkg.keys()
+        assert pkg.meta == {'name': 'test_meta'}
+        existing_one_entry = pkg['nested/one.txt']
+
+        nested_dir = pathlib.Path(data_dir, "nested2")
+        nested_dir.mkdir(parents=True, exist_ok=True)
+        with open(nested_dir / 'one.txt', 'w') as fd:
+            fd.write(fd.name)
+
+        # existing update policy
+        pkg = pkg.set_dir("nested", nested_dir, update_policy='existing')
+        assert 'nested' in pkg.keys()
+        assert existing_one_entry == pkg['nested/one.txt']
+
+        # incoming update policy
+        pkg = pkg.set_dir("nested", nested_dir)
+        assert 'nested' in pkg.keys()
+        assert existing_one_entry != pkg['nested/one.txt']
+
+        # verify non existing update policy raises value error
+        expected_err = f"Update policy should be one of {str(PACKAGE_UPDATE_POLICY)}, not 'invalid_policy'"
+        with pytest.raises(ValueError) as e:
+            pkg.set_dir("nested", data_dir, update_policy='invalid_policy')
+        assert expected_err in str(e.value)
+
+        # remove created directory
```

Comment:
same here; let's not modify the data directory inside the test (again you can add data outside the test) and/or change directories, etc.